src/Microsoft.AspNetCore.Hosting/WebHostExtensions.cs

```diff
@@ -90,8 +89,7 @@ public static void Run(this IWebHost host)
                     done.Wait();
                 };
 
-                var assemblyLoadContext = AssemblyLoadContext.GetLoadContext(typeof(WebHostExtensions).GetTypeInfo().Assembly);
-                assemblyLoadContext.Unloading += context => shutdown();
+                AppDomain.CurrentDomain.ProcessExit += (sender, eventArgs) => shutdown();
```

Comment:
Can we make this change in a follow up if we can't get a sign off? There wasn't a test written for this as part of the changeset this was introduced in so we might not be capturing regressions in behavior here.